e2e/e2e_suite_test.go

```diff
@@ -0,0 +1,76 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/go-logr/logr"
+	"github.com/medik8s/machine-deletion/api/v1alpha1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap/zapcore"
+	"k8s.io/api/apps/v1beta1"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	ctrl "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	log           logr.Logger
+	dynamicClient dynamic.Interface
+	clientSet     *kubernetes.Clientset
+	k8sClient     ctrl.Client
+
+	// The ns test pods are started in
+	testNsName = "mdr-test"
+)
+
+func TestE2e(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "E2e Suite")
+}
+
+var _ = BeforeSuite(func() {
+	opts := zap.Options{
+		Development: true,
+		TimeEncoder: zapcore.RFC3339NanoTimeEncoder,
+	}
+	logf.SetLogger(zap.New(zap.WriteTo(GinkgoWriter), zap.UseFlagOptions(&opts)))
+	log = logf.Log
+
+	// +kubebuilder:scaffold:scheme
+
+	// get the k8sClient or die
+	config, err := config.GetConfig()
+	if err != nil {
+		Fail(fmt.Sprintf("Couldn't get kubeconfig %v", err))
+	}
+	clientSet, err = kubernetes.NewForConfig(config)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(clientSet).NotTo(BeNil())
+
+	dynamicClient, err = dynamic.NewForConfig(config)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(dynamicClient).NotTo(BeNil())
+
+	scheme.AddToScheme(scheme.Scheme)
+	err = v1alpha1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+	err = v1beta1.AddToScheme(scheme.Scheme)
+	Expect(err).NotTo(HaveOccurred())
+
+	k8sClient, err = ctrl.New(config, ctrl.Options{Scheme: scheme.Scheme})
+	Expect(err).NotTo(HaveOccurred())
+
+	debug()
+})
+
+func debug() {
```

Comment:
Nit: I'm not sure if this was meant to be kept, in case so maybe use better name

Reply:
I took it from another operator, but indeed the name isn't the best